1.8

dmarcuse opened the pull request:
To do before merging:
- [ ] Better config system (w/ GUI preferably)
- [x] Better custom ROM system, with plugin API
- [ ] Computer metadata system, for storing arbitrary data (e.g. labels and attached peripherals)

Lustyn committed:
Update to 1.8

Commit:
Merge pull request #63 from justync7/1.8

Update to 1.8

Lustyn committed:
Fix colour clamping and setup logger

Lustyn committed:
Add palette caching to prevent artifacting

SquidDev committed:
Update to the latest build of ComputerCraft

Commit:
Merge pull request #64 from SquidDev-CC/1.8

Update to the latest build of ComputerCraft

dmarcuse committed:
Fix #66

SquidDev committed:
Add some basic hooks for various terminal actions

Commit:
Merge pull request #70 from SquidDev-CC/feature/terminal-hooks

Add hooks to EmulatedTerminal

SquidDev committed:
Add TRoR renderer

dmarcuse committed:
Add Lombok, switch logging backend to log4j

dmarcuse committed:
Workaround for CCEmuX.getVersion failing

dmarcuse committed:
Remove Renderer.resize

dmarcuse committed:
Bump Gradle version

dmarcuse committed:
Bump dependency versions

dmarcuse committed:
Tidy up TerminalFont

SquidDev committed:
Replace COLOURS constant with one in Utils

SquidDev committed:
Make InputPacket package private

dmarcuse committed:
Use ConcurrentHashMap to track EmulatedComputer instances

SquidDev committed:
Add handling for computers being closed, shutdown and resized

SquidDev committed:
Add support for changing the TRoR visibility

This will broadcast the terminal state when shown.

Commit:
Merge pull request #71 from SquidDev-CC/feature/tror-renderer

Add basic TRoR renderer

dmarcuse committed:
Add drag-and-drop support for files and text to AWTRenderer

dmarcuse committed:
Merge branch '1.8' of git@github.com:Lignum/CCEmuX.git into 1.8

dmarcuse committed:
Add alert for successful file copy, check size before copying

dmarcuse committed:
Track plugin sources

dmarcuse committed:
Revamp TerminalFont, add HD font plugin (Thanks @BombBloke!)

dmarcuse committed:
Fix typo, throw RuntimeException when no fonts available

dmarcuse committed:
Add VirtualMount and related code

dmarcuse committed:
Add CreatingROM hook, remove CustomRomMount

dmarcuse committed:
Port ccemux API and custom ROM to built-in plugin

dmarcuse committed:
Improve credits

dmarcuse committed:
Fix missing test dependencies

dmarcuse committed:
Fix test dependencies again because I'm dumb

Lignum commented on src/main/java/net/clgd/ccemux/Utils.java:
```diff
@@ -22,6 +29,20 @@ public static Color getCCColourFromChar(char c) {
+		return new double[]{clamp(col[0], 0f, 1f), clamp(col[1], 0f, 1f), clamp(col[2], 0f, 1f)};
+	}
+
+	public static double clamp(double f, double min, double max) {
```
`return Math.min(Math.max(f, min), max);` is a lot nicer

dmarcuse committed:
Improve Utils.clampColor

dmarcuse committed:
Use AutoService for service file

dmarcuse committed:
Patch -r argument

dmarcuse committed:
Bump Gradle to 4.0.1

dmarcuse committed:
Fix NPE when splash screen not present

dmarcuse committed:
New animated splash screen

dmarcuse committed:
Move image resources to new folder

Lignum committed:
Fix palette change delay

dmarcuse committed:
Fix crash in headless environments

dmarcuse committed:
Fix tab not always working

dmarcuse committed:
Remove leftover logging statement

dmarcuse committed:
Fix memory leak with tinted font images

dmarcuse committed:
Improve crash reporter

dmarcuse committed:
Replace CCEmuX.addComputer with CCEmuX.createComputer

This allows callers to pass a Consumer that will be called on the
EmulatedComputer.Builder object

dmarcuse committed:
Add new config framework

dmarcuse committed:
Add extra config handler test

dmarcuse committed:
Package CC with CCEmuX

dmarcuse committed:
Update to latest CC (CCTweaks disabled, will enable once updated)

dmarcuse committed:
Add license info

SquidDev committed:
Bump CCTweaks-Lua version

Also fix CCEmuX failing on Windows due to paths using '\' instead of '/',
thus meaning the bios could not be found.

dmarcuse committed:
Remove --cc argument, separate jar tasks and add "no-cc" jar

dmarcuse committed:
Update README

dmarcuse committed:
Remove unused dependency

dmarcuse committed:
Custom classloader can be disabled via ccemux.forceDirectLaunch property

dmarcuse committed:
Add lombok config to fix JDK 9 compilation
